pkg/provisioner/ironic/ironic.go

```diff
@@ -230,6 +230,25 @@ func (p *ironicProvisioner) validateNode(ironicNode *nodes.Node) (errorMessage s
 	return "", nil
 }
 
+func (p *ironicProvisioner) listAllPorts(address string) ([]ports.Port, error) {
+	var allPorts []ports.Port
+
+	opts := ports.ListOpts{}
+
+	if address != "" {
+		opts.Address = address
+	}
+
+	allPages, err := ports.List(p.client, opts).AllPages()
```

Comment:
List returns a pagination.Pager that could have an error (pager.Err), it could be nice to check it before gettings all the pages

Reply:
Yes, `ports.List()` returns a `pagination.Pager` but it's simply an instance of the type with nothing in it. No work has been done yet. It doesn't touch the network until you try to paginate (or access the results), and thus there are no errors yet. 

https://github.com/gophercloud/gophercloud/blob/eeff63dbdcab503e85178c5faa8a0acd4735f7f7/openstack/baremetal/v1/ports/requests.go#L67

https://github.com/gophercloud/gophercloud/blob/eeff63dbdcab503e85178c5faa8a0acd4735f7f7/pagination/pager.go#L54

Dealing with the `err` returned from `AllPages()` should be enough here.

Reply:
Mmh, I think there could be a case of error (https://github.com/gophercloud/gophercloud/blob/eeff63dbdcab503e85178c5faa8a0acd4735f7f7/openstack/baremetal/v1/ports/requests.go#L63).

I mentioned it because I did the same mistake in another point of the code using a similar pattern, and ignoring the pager error made the troubleshooting really a pain.

Reply:
Good point, fixed!